lib/grid.rb

```diff
@@ -0,0 +1,39 @@
+require_relative './grid_creator'
+
+class Grid
+
+  include GridCreator
+  attr_reader :cells, :rows, :columns, :boxes, :unsolved_cells
+
+  def initialize(puzzle)
+    @cells = puzzle.split('').map { |v| Cell.new(v) }
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.